tests/hikari/impl/test_rate_limits.py

```diff
@@ -401,6 +402,33 @@ def test_is_rate_limited_when_rate_limit_not_expired_only_returns_False(self, re
 
 
 class TestExponentialBackOff:
+    def test___init___raises_on_too_large_int_base(self):
+        base = int(sys.float_info.max) + int(sys.float_info.max * 1 / 100)
+        with pytest.raises(ValueError, match="int too large to be represented as a float"):
```

Comment:
why / 100, / 200, / 300?

Reply:
No particular reason, just wanted a number that was different enough to actually be different after rounding and didn't want to duplicate data between test cases

Reply:
but... why would it overflow if you did less than `float.max * random() / 300` of the maximum?

surely it needs to be >= float.max to do that?

Reply:
Because the result of that is being added onto the max?

Reply:
ah ok, was being stupid, sorry